core/src/main/scala/frameless/CatalystSummable.scala

```diff
@@ -0,0 +1,26 @@
+package frameless
+
+/**
+  * When summing Spark doesn't change these types:
+  * - Long       -> Long
+  * - BigDecimal -> BigDecimal
+  * - Double     -> Double
+  *
+  * For other types there are conversions:
+  * - Int        -> Long
+  * - ...
+  */
+trait CatalystSummable[A] {
+  def zero: A
+}
+
+object CatalystSummable {
+  def apply[A](zero: A): CatalystSummable[A] = {
+    val _zero = zero
+    new CatalystSummable[A] { val zero: A = _zero }
+  }
+
```

Comment:
Right, but `CatalystSummable` is also used for aggregate operation, `plus` is called as `Add` in Spark, I will create typeclase CatalystAddable for it.